integration/store_gateway_limits_hit_test.go

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+//go:build requires_docker
+// +build requires_docker
+
+package integration
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/grafana/e2e"
+	e2ecache "github.com/grafana/e2e/cache"
+	e2edb "github.com/grafana/e2e/db"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/storegateway"
+
+	"github.com/grafana/mimir/pkg/util/globalerror"
+
+	"github.com/grafana/mimir/integration/e2emimir"
+)
+
+const (
+	blockRangePeriod = 5 * time.Second
+	ingesterTag      = "ingester"
+	distributorTag   = "distributor"
+	storeGatewayTag  = "store-gateway"
+	querierTag       = "querier"
+	orgID            = "test"
+	series1          = "series_1"
+	series2          = "series_2"
+	series3          = "series_3"
+	series4          = "series_4"
```

Comment:
[nit] I would remove all these constants. Don't think give much value, if you accept the comment I left on `createClient()`.

Reply:
I did accept the comment about `createClient()` and removed some of the constants. I have left a couple of them that are used more times, in order to avoid hardcoding.